test/cql-pytest/test_null.py

```diff
@@ -65,6 +79,51 @@ def test_insert_null_key_lwt(cql, table1):
     with pytest.raises(InvalidRequest, match='null value'):
         cql.execute(stmt, [None, s])
 
+# INSERT statements must specify all components of the primary key,
+# and the corresponding query parameters, if any, must not be null.
+# If INSERT or DELETE statement is missing a non-last element of
+# the primary key, the error message generated contained
+# an invalid column name (#12046).
+# The problem occurred if the query contains a column with the list type,
+# otherwise statement_restrictions::process_clustering_columns_restrictions
+# checks that all the components of the key are specified.
+def test_insert_with_list_column_and_missing_clustering_key_part(cql, table2):
+    key = unique_key_string()
+    with pytest.raises(InvalidRequest,
+                       match='Missing mandatory PRIMARY KEY part a|Some clustering keys are missing: a'):
+        cql.execute(cql.prepare(f"INSERT INTO {table2} (id,b,c) VALUES ('{key}',1,null)"))
```

Comment:
If I understand correctly, the important part of this test is that this INSERT is missing a value for "a".
So how does the fact that "c" - some non-key column - is assigned a null - related to this test? Is it really related? Assigning null to c, which is a list, is perfectly fine - it just assigns an empty list.

Reply:
The important part is not null assigned to list, but that the query just contains a list column. The reason is described above.